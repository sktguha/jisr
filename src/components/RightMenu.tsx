import React, { useEffect } from 'react';
import { genericFn } from '../types/App.types';

export function RightMenu({ onAddNewFolder, onAddNewFile, fileName, onDelete, onRename, setIsRightMenu, isFile }: {
    onAddNewFolder?: genericFn; onAddNewFile?: genericFn; fileName: string | undefined; onDelete: genericFn; onRename: genericFn; setIsRightMenu: genericFn;
    isFile?: boolean;
}) {
    useEffect(() => {
        document.getElementById("right-menu-item")?.focus();
    }, []);
    return (
        <div 
            tabIndex={-1}
            className="right-menu-main" id="right-menu-item"
            onBlur={() => setIsRightMenu(false)}
        >
            {/* RIGHT MENU FUNCTIONS {fileName} */}
            <div onClick={() => { setIsRightMenu(false); }}>
                {!isFile && <div className="menu-item" onClick={() => { (onAddNewFolder as genericFn)(); setIsRightMenu(false); }}>Add New Folder</div>}
                {!isFile && <div className="menu-item" onClick={() => { (onAddNewFile as genericFn)(); setIsRightMenu(false); }}>Add New File</div>}
                <div className="menu-item" onClick={(e) => {
                    console.log(e, fileName);
                    setIsRightMenu(false);
                }}>Copy ( only logs to console)</div>
                <div className="menu-item" onClick={() => { onDelete(); setIsRightMenu(false); }}>Delete</div>
                <div className="menu-item-last" onClick={() => { onRename(); setIsRightMenu(false); }}>Rename</div>
            </div>
        </div>
    );
}
