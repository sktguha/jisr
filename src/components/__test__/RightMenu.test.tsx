import React from 'react';
import { render, screen } from '@testing-library/react';
import {RightMenu} from '../RightMenu';

const emptyFn = ()=>{};

test('renders RightMenu folder', () => {
  const { container } = render(<RightMenu setIsRightMenu={emptyFn} onAddNewFile={emptyFn} onRename={emptyFn} fileName='image.webp' onDelete={emptyFn}/>)
  expect(container).toMatchSnapshot()
});

test('renders RightMenu file', () => {
    const { container } = render(<RightMenu isFile setIsRightMenu={emptyFn} onAddNewFile={emptyFn} onRename={emptyFn} fileName='image.webp' onDelete={emptyFn}/>)
    expect(container).toMatchSnapshot()
  });
  