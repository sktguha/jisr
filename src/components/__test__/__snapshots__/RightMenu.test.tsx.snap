// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders RightMenu file 1`] = `
<div>
  <div
    class="right-menu-main"
    id="right-menu-item"
    tabindex="-1"
  >
    <div>
      <div
        class="menu-item"
      >
        Copy ( only logs to console)
      </div>
      <div
        class="menu-item"
      >
        Delete
      </div>
      <div
        class="menu-item-last"
      >
        Rename
      </div>
    </div>
  </div>
</div>
`;

exports[`renders RightMenu folder 1`] = `
<div>
  <div
    class="right-menu-main"
    id="right-menu-item"
    tabindex="-1"
  >
    <div>
      <div
        class="menu-item"
      >
        Add New Folder
      </div>
      <div
        class="menu-item"
      >
        Add New File
      </div>
      <div
        class="menu-item"
      >
        Copy ( only logs to console)
      </div>
      <div
        class="menu-item"
      >
        Delete
      </div>
      <div
        class="menu-item-last"
      >
        Rename
      </div>
    </div>
  </div>
</div>
`;
